README.md

```diff
@@ -78,3 +78,66 @@ module "teleport_bootstrap_script" {
   environment = "${terraform.workspace}"
 }
 ```
+
+## teleport-ecs
+
+This module will deploy Teleport on ECS. This takes care of the `auth` and `proxy` components.
+
+These are the different ports that Teleport exposes, and the LB that they are attached to.
+
+- `auth` port 3025: Used by nodes to register with the cluster. Mapped to the same port in the private NLB (ARN provided by parameter to this module).
+- `proxy` port 3023: Used for clients to SSH into a node. Mapped to the same port in the public ELB.
+- `proxy` port 3024: Used for other Teleport clusters to open a reverse tunnel (trusted clusters). Mapped to the same port in the public ELB.
+- `proxy` port 3080: Used to serve the Web UI. Mapped to port 443 in the public ELB.
+
+Audit logs generated in the `auth` server will be shipped to CloudWatch logs, it'll create a new log group named `"teleport_logs_${var.environment}_${var.project}"`.
+
+### Available variables:
+* [`cluster_name`]: String(required): Name of the cluster.
+* [`domain_name`]: String(required): Domain name of where we want to reach our cluster. Example can be `company.com`
+* [`nlb_private_arn`]: String(required): ARN for the private NLB to create a listener for the Node `auth` containers
+* [`vpc_id`]: String(required): VPC ID of where we want to deploy Teleport in
+* [`aws_region`]: String(optional): AWS region where the CloudWatch logs are located. Defaults to eu-west-1
+* [`cpu`]: Integer(optional): The number of CPU units used by the task. It can be expressed as an integer using CPU units, for example 1024, or as a string using vCPUs, for example 1 vCPU or 1 vcpu, in a task definition but will be converted to an integer indicating the CPU units when the task definition is registered. Defaults to 128.
+* [`dynamodb_table`]: String(optional): Which DynamoDB table does teleport need, teleport will create this table for you. You don't need to define anything in Terraform. Defaults to main.teleport
+* [`dynamodb_region`]: String(optional): In which region does the DynamoDB table need to be created. Defaults to eu-west-1
+* [`environment`]: String(optional): Environment where this node belongs to, will be the third part of the node name. Defaults to ''
```

Comment:
can't you fetch this from terraform.workspace? afaik you have access to that inside the module.

Reply:
I wouldn't do it like that, as we would be limiting this module to a specific use case. We do map environments to workspaces, but not everybody does, even we don't do it everywhere.
We should keep our terraform modules as generic as possible imo